### ISSUE TITLE
Feature request: .png files should be re-compressed with highest compression ratio to save space and bandwidth

### ISSUE THREAD

**Issue by partev** [yairm210/Unciv#11890]
### Before creating

- [X] This is NOT a gameplay feature from Civ VI, BNW, or outside - see Roadmap
- [X] This is NOT a gameplay feature from Vanilla Civ V or from G&K - If so, it should be a comment in https://github.com/yairm210/Unciv/issues/4697

### Problem Description

.PNG files can be re-compressed to save significant storage space and bandwidth.
For example I re-compressed [Feature graphic - Google Play.png](https://github.com/yairm210/Unciv/blob/master/extraImages/Feature%20graphic%20-%20Google%20Play.png) and got about 50k of saving:

`$ advpng -z -4 Feature\ graphic\ -\ Google\ Play.png`
`      896269      841727  93% Feature graphic - Google Play.png`

### Desired Solution

run "advpng -z -4 -i 100" on all .png files in the repository

### Alternative Approaches

do nothing

**Comment by yairm210** [comment]
That's really cool! 
Sounds like something we can add as part of our pipeline 👍🏿 
Not sure if we can provide this for mods, will require research

**Comment by yairm210** [comment]
Unfortunately this doesn't seem to be available as a public tool in any recent, it's part of advanceCOMP 
Did manage to find a 2.1 version available via `apt install advcomp`
Ran on flags with `ls -1 | xargs advpng -z -4` - seemed to go well

To run on all files I used
`find . -type f -regex ".*png" | xargs -d '\n' advpng -z -4`

Unfortunately this doesn't seem to shrink the file sizes of the *compiled pngs* built out of the smaller ones - but when ran on the final game.pngs, it does decrease the size. This is only like a 14% size decrease, which isn't super great but it's better than nothing.

To really have a long term impact we'll need to be able to make changes in the repo itself... sounds like "git checkout, install advcomp, run advpng, if any dif - git commit" to me, sounds doable

**Comment by yairm210** [comment]
`find . -type f -regex ".*png" | grep -v mods | xargs -d '\n' advpng -z -4` for not processing mods...
Around 22% savings for compiled pngs

Was very slow, though, so looking online for other lossless compression tools I found [oxipng](https://github.com/shssoichiro/oxipng)
And DAMN is it fast, and with 
`find . -type f -regex ".*png" | grep -v mods | xargs -d '\n' oxipng -o 4`
managed to eat through 2,200 files in a couple of minutes

I'm still unclear on how to add this to the pipeline, but I guess if it's only on the compiled pngs... then I don't really need to commit it anyway, I can just compress, continue with the regular release cycle, everyone gets slightly smaller files and we're done

**Comment by touhidurrr** [comment]
I wonder if this would really have any effect on anything game related. Bad that I did not see it before merging. My reasons are as following:
1. Based on my understanding of how Unciv works, images gets merged together in some files and an atlas file is generated for them. So, any compression on the original files should have no effect on what goes inside the app.
2. Although future images can be losslessly compressed before adding to repo. Data of any file ever added via a commit always remains in git history, all of which are needed to be fetched every time someone clones it. So, compressing everything and uploading them now only increases the bandwidth used now.

Again, I can be wrong. But based on my understanding regarding this matter, I see no point in it. I personally use a lossless compression software called [Caesium Image Compressor](https://github.com/Lymphatus/caesium-image-compressor) on a regular basis but mostly to before uploading stuff to cloud drives. With `.git`, once an image is added, it makes little sense to me to compress it again to me.

**Comment by touhidurrr** [comment]
So, curious if this actually reduced `.jar` or `.apk` size.

**Comment by touhidurrr** [comment]
Also, @yairm210, can you update the atlas and generated png files also? when i tried to run `./gradlew.bat desktop:run` to fix my merge conflict with master (for Flags.png), it tried to generate lots of png and atlas files and i committed them by kind of mistake. Later needed to force reset one commit.

**Comment by yairm210** [comment]
A. This should make a difference for files not packed, like tutorials and everything under extra images
https://github.com/yairm210/Unciv/tree/master/android/assets/ExtraImages

B. I see... Then we do need to introduce this in the build if we want to minimize final file size for releases :/
How annoying

**Comment by touhidurrr** [comment]
![image](https://github.com/yairm210/Unciv/assets/46617994/541ac688-088c-48d8-ba9d-b413420d0099)

I am curious why size of every generated `.png` file increased even though the size of originals decreased. This aligns with my original assumption that generated files size are unrelated to the size of originals, but more than 100% ? Weird.

**Comment by yairm210** [comment]
Nono, when I compressed all files I also compressed the generated ones...
When "recreating them from scratch" they returned to their pre-compressed sizes (generating pngs is basically dealing directly with bitmaps so it doesn't care about compression of constituents)

**Comment by touhidurrr** [comment]
> 
Nono, when I compressed all files I also compressed the generated ones...
When "recreating them from scratch" they returned to their pre-compressed sizes (generating pngs is basically dealing directly with bitmaps so it doesn't care about compression of constituents)

I wonder how does the game detects and write outdated png files. If we compress a generated file and ship it, will the game somehow regenerate all of them and will it cause a initial lag for players i wonder. But do they even have the access to the originals to generate them to begin with, thats also something that comes to my mind.

In the end, whether to lossless compress or not to, looks like a really complicated topic.
